50_wa_diff_overdose.py

```diff
@@ -0,0 +1,299 @@
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+import seaborn as sns
+import altair as alt
+from plotnine import *
+import statsmodels.api as sm
+import statsmodels.formula.api as smf
+
+# load in pre-cleaned deaths data for Washington and comparison states
+deaths = pd.read_csv("05_cleaned_data/deaths_wa.csv")
+# two separate dfs - one for washington and one for comp states
+washington = deaths[deaths["StateName"] == "Washington"]
+# comp = deaths[deaths["StateName"] != "Washington"]
+comp = deaths[
+    (deaths["StateName"] == "Oregon")
+    | (deaths["StateName"] == "Idaho")
+    | (deaths["StateName"] == "Montana")
+    | (deaths["StateName"] == "Nevada")
+    | (deaths["StateName"] == "Wyoming")
+]
+
+
+def select_overdose(record):
+    """Simple function to select only overdose records"""
+
+    if record == "All other non-drug and non-alcohol causes":
+        return 0
+
+    if record == "All other alcohol-induced causes":
+        return 0
+
+    if record == "All other drug-induced causes":
+        return 0
+
+    if record == "Alcohol poisonings (overdose) (X45, X65, Y15)":
+        return 0
+
+    if record == "Drug poisonings (overdose) Unintentional (X40-X44)":
+        return 1
+
+    if record == "Drug poisonings (overdose) Suicide (X60-X64)":
+        return 1
+
+    if record == "Drug poisonings (overdose) Undetermined (Y10-Y14)":
+        return 1
+
+    else:
+        return "error"
+
+
+# copy to fix the dreaded "A value is trying to be set on a copy of a slice" error
+wa_deaths = washington.copy()
+comp_deaths = comp.copy()
+
+# apply new function to our df
+wa_deaths["overdose"] = wa_deaths["Drug/Alcohol Induced Cause"].apply(
+    lambda x: select_overdose(x)
+)
+comp_deaths["overdose"] = comp_deaths["Drug/Alcohol Induced Cause"].apply(
+    lambda x: select_overdose(x)
+)
+
+# filter accordingly based on new column
+wa_deaths = wa_deaths[wa_deaths["overdose"] != 0]
+comp_deaths = comp_deaths[comp_deaths["overdose"] != 0]
+
```

Comment:
In continuation with with the comments on line 24 - 49. This is a simpler method for creating the subsets for drug overdose deaths for Washington and comparison states. 

Creating the subset for Washington which has all drug related overdose deaths 

wa_deaths = washington[
    (
        (
            washington["Drug/Alcohol Induced Cause"]
            == "Drug poisonings (overdose) Unintentional (X40-X44)"
        )
        | (
            washington["Drug/Alcohol Induced Cause"]
            == "Drug poisonings (overdose) Undetermined (Y10-Y14)"
        )
        | (
            washington["Drug/Alcohol Induced Cause"]
            == "Drug poisonings (overdose) Suicide (X60-X64)"
        )
    )
]

Creating the subset for comparison states for Washington which has all drug related overdose deaths 

comp_deaths = comp[
    (
        (
            comp["Drug/Alcohol Induced Cause"]
            == "Drug poisonings (overdose) Unintentional (X40-X44)"
        )
        | (
            comp["Drug/Alcohol Induced Cause"]
            == "Drug poisonings (overdose) Undetermined (Y10-Y14)"
        )
        | (
            comp["Drug/Alcohol Induced Cause"]
            == "Drug poisonings (overdose) Suicide (X60-X64)"
        )
    )
]

Reply:
Thanks for providing an easier way to solve this problem!